File: Easy/1572.cpp
// 1572. Matrix Diagonal Sum

class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int primary = 0,secondary = 0;
        for(int i=0;i<mat.size();i++){
            primary += mat[i][i];
            secondary += mat[i][mat.size()-1-i];
        }
        if(mat.size()%2!=0){
            int mid = mat.size()/2;
            secondary -= mat[mid][mid];
        }
        return primary+secondary;        
    }
};

File: Easy/1572.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int r = mat.length; // calculate length
        int c = r; // calculate columns -> bcs its a square matrix
        int ans = 0; // total sum

        // primary
        for(int i=0;i<r;i++){
            ans += mat[i][i]; 
        }

        //secondary
        for(int i=0;i<r;i++){
            if(i==(c-i-1)){ 
                ans+=0;
            }else{
                ans+=mat[i][c-i-1];
            }        
        }
        return ans;
    }
}
